### PR TITLE
PYTHON-941 Improve error for batch WriteTimeouts

### PR DESCRIPTION
**Pull request by bjmb**
Many changes just to add something an error message, but we were getting circular import errors if we leaved `WriteType` in policies

### CODE REVIEW

File: cassandra/__init__.py [hunk 1]
```diff
@@ -339,14 +409,16 @@ class Timeout(RequestExecutionException):
...
 
-    def __init__(self, summary_message, consistency=None, required_responses=None, received_responses=None):
-        self.consistency = consistency
-        self.required_responses = required_responses
-        self.received_responses = received_responses
-        Exception.__init__(self, summary_message + ' info=' +
-                           repr({'consistency': consistency_value_to_name(consistency),
-                                 'required_responses': required_responses,
-                                 'received_responses': received_responses}))
+    def __init__(self, summary_message, **kwargs):
```
[the remark this answers is not in the document]
Comment (bjmb) [reply cassandra/__init__.py]: Yes, I'm :+1: to leaving the arguments, however `Timeout` is the parent of `WriteTimeout` and `ReadTimeout` so I'd rather not add a parameter in `Timeout` which is `write_type`, which is specific to writes only.
I've added `**kwargs` after the arguments to allow child classes to add more parameters to the exception message, in our case we use for `WriteTimeout` to add `write_type`

**Comment by mambocab** [comment]
Cool, API change looks good -- while I had it pulled down to run the tests locally I added a changelog entry and changed the test so it would require the exception be raised, rather than passing.

+1, if the tests run clean and you're happy, merge away!

**Comment by bjmb** [comment]
Awesome, thanks for the review!

File: cassandra/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mambocab) [reply cassandra/__init__.py]: Missed this on my re-review, thanks for the explanation!

**Pull request merged**